addons/website/models/website.py

```diff
@@ -341,23 +347,70 @@ def set_colors(selected_palette):
+            menu_count = len(features.filtered('in_menu'))
+            company_menu_count = len(features.filtered('company_menu'))
+
+            unlink_menu_domain = []
+            company_menu = False
+            if menu_count > 5 and company_menu_count > 1:
+                company_menu = self.env['website.menu'].create({
+                    'name': _('Company'),
+                    'parent_id': website.menu_id.id,
+                    'website_id': website.id,
+                    'sequence': 40,
+                })
+                unlink_menu_domain.append(('id', '!=', company_menu.id))
+
             pages_views = {}
             modules = self.env['ir.module.module']
-            for feature_id in feature_ids:
-                if feature_id.type == 'app' and feature_id.module_id and feature_id.module_id.state != 'installed':
-                    modules += feature_id.module_id
-                if feature_id.type == 'page' and feature_id.page_view_id:
-                    result = self.env['website'].new_page(name=feature_id.name, add_menu=True, template=feature_id.page_view_id.key)
-                    pages_views[feature_id.iap_page_code] = result['view_id']
+            module_data = {}
+            footer_links = []
+            for feature in features:
+                if feature.type == 'app' and feature.module_id:
+                    if feature.module_id.state != 'installed':
+                        modules += feature.module_id
+                    if feature.in_menu:
+                        if feature.url != '/blog':
+                            module_data[feature.url] = {'sequence': feature.menu_sequence}
+                        else:
+                            blogs = module_data.setdefault('/blog', [])
+                            blogs.append({'name': feature.name, 'sequence': feature.menu_sequence})
+                if feature.type == 'page' and feature.page_view_id:
+                    page_values = {
+                        'name': feature.name,
+                        'url': feature.url,
+                        'is_published': True,
+                    }
+                    menu_values = {
+                        'name': feature.name,
+                        'url': feature.url,
+                        'sequence': feature.menu_sequence,
+                        'parent_id': company_menu.id if company_menu and feature.company_menu else website.menu_id.id,
+                    }
+                    result = self.env['website'].new_page(name=feature.name, add_menu=feature.in_menu, template=feature.page_view_id.key)
+                    page = self.env['ir.ui.view'].browse(result['view_id']).page_ids
+                    page.write(page_values)
+                    if feature.in_menu:
+                        menu = page.menu_ids
+                        menu.write(menu_values)
+                    pages_views[feature.iap_page_code] = result['view_id']
```

Comment:
Alternative: update `new_page` by adding some parameters in its signature

Reply:
oui ca serait mieux.
Vu qu'en v17 on admettra que même sans le configurator faire une 'new page' avec un template 'pricing page' sera cool ;)

d'ailleur la notion existe de template existe deja mais correspond a la key d'une autre vue. Mai son pourrait fallbacker sur un hasattr de website pour generer ce type de page. Qu'en penses tu ?

Reply:
Je comprends pas l'idée de falback sur un hasattr. Enfin je vois pas comment ca fonctionnerait

Reply:
new_page(template=''website.default_page')

```
if view.ref(template):
   // use pricing as view template
else: -> le fallback
   if '_new_page_%s' % template) in website:  -> le hasattr
      // create your custom pricing page
```